src/content/en/updates/2019/08/barcode-detection.md

```diff
@@ -0,0 +1,97 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: The Barcode Detection API allows for barcodes to be detected in images.
+
+{# wf_updated_on: 2019-08-02 #}
+{# wf_published_on: 2019-08-01 #}
+{# wf_tags: capabilities,shape-detection,progressive-web-apps,webapp #}
+{# wf_featured_image: /web/updates/images/generic/timeline.png #}
+{# wf_featured_snippet: The Barcode Detection API allows for barcodes to be detected in images. #}
+{# wf_blink_components: Blink>ImageCapture #}
+
+# Shipping the Barcode Detection Part of the Shape Detection API {: .page-title}
+
+{% include "web/_shared/contributors/thomassteiner.html" %}
+
+<div class="clearfix"></div>
+
+You may remember [my post from January](../01/shape-detection)
+on the [Shape Detection API](https://wicg.github.io/shape-detection-api),
+an API that defines three interfaces: the `BarcodeDetector`, the `FaceDetector`,
+and the `TextDetector` interface. I'm now happy to share that following a successful Origin Trial
+we're shipping the `BarcodeDetector` part of the Shape Detection API in Chrome&nbsp;78.
+The `FaceDetector` and the `TextDetector` interface still need some more fine-tuning.
+
+### Working with the `BarcodeDetector` {: #barcodedetector}
+
+Let's briefly recall how working with the `BarcodeDetector` looks like:
+
+```js
+const barcodeDetector = new BarcodeDetector();
+try {
+  const barcodes = await barcodeDetector.detect(image);
+  barcodes.forEach(barcode => console.log(barcode));
+} catch (e) {
+  console.error('Barcode detection failed:', e);
+}
+```
+
+### Origin Trial Feedback Summary
+
+The primary feedback during the Origin Trial was around the limited availability
+of barcode detection support across Blink platforms.
+In response to this feedback, we have now added a new static `getSupportedFormats()` method
+to more easily allow developers to detect when the current platform supports
+the desired formats and when a polyfill is necessary.
+The code sample below shows how this new method could be used to detect QR codes
+(among several
+[other possible barcode formats](https://wicg.github.io/shape-detection-api/#enumdef-barcodeformat)).
+
+```js
+// 🆕 Check if the platform supports QR codes
+if ((await BarcodeDetector.getSupportedFormats()).includes('qr_code')) {
+  try {
+    // If it does, try to decode the QR code
+    const detectedCodes = await (new BarcodeDetector()).detect(someImage, {
+      formats: ['qr_code']
+    });
```

Comment:
The `formats` option is passed to the `BarcodeDetector` constructor, not to the `detect()` method. This code should be,

```javascript
const detectedCodes = await (new BarcodeDetector({ formats: ['qr_code'] })).detect(someImage);
```

However, we should avoid recommending developers create a new `BarcodeDetector` for every detection. As the specification mentions, this could be an expensive operation.

During the Origin Trial the format hinting parameter wasn't hooked up to the implementation. Can you add some text explaining that providing this hint allows the implementation to optimize for detecting only the formats specified?

Reply:
Thanks, fixed.